lib/src/atom/matcher.rs

```diff
@@ -150,38 +155,52 @@ impl Bindings {
         }
     }
 
-    fn add_var_equality(mut self, a: &VariableAtom, b: &VariableAtom) -> Vec<Bindings> {
-        match (self.id_by_var.get(a).copied(), self.id_by_var.get(b).copied()) {
-            (Some(a_var_id), Some(b_var_id))  =>
+    /// Asserts equality between two [VariableAtom]s.  If the existing bindings for `a` and `b` are
+    /// incompatible then this method will return an error.  This method will also return an error if
+    /// this operation causes split bindings.  If split bindings are a possibility, call
+    /// [BindingsSet::add_var_equality] instead.
+    pub fn add_var_equality(self, a: &VariableAtom, b: &VariableAtom) -> Result<Bindings, Atom> {
```

Comment:
I am sorry for being unclear. I meant that we could get rid of `.unwrap()` by using signatures like: `fn add_var_equality(self, a: &VariableAtom, b: &VariableAtom) -> Bindings` and do `panic!` inside but didn't said this explicitly (returning `Bindings` means that `add_var_binding()` method should use `add_var_binding_internal()` inside and check result itself). I was not 100% sure we need this, because I like explicit errors as well. But it seems for these methods user knows what he is doing by calling them, so `panic!` can be a proper way to notify him about error.

Now I think we can just do step forward and return `Bindings` or step back and return `Result<Bindings, String>`. I would do the first but if you don't like it please do whatever seems right to you.

Reply:
Ah ok.  I misunderstood.  IMO the `?` operator makes returning `Result` very nice.  But it's definitely more overhead than just `panic!`.

So my personal preference is to return Result<>, although that's my personal preference and I feel like your preferences deserve a stronger weight in this context.

Reply:
I pushed a change to convert the functions to return `Result<Bindings, &'static str>`.  My feeling is that it's conceivable there is a use case for constructing individual Bindings where we might not want to panic!, but the author might not be able to guarantee the validity of the inputs in advance.  For example, an interactive MeTTa editor, where invalid bindings might be applied, temporarily.

Reply:
Yes, sure, using `?` looks good, and I agree being explicit about errors works better in API. Thanks!